### PR TITLE
feat: create directories for history if they DNE

### LINKED ISSUE #385: try to create directories for history if they don't already exist

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #385.

### CODE REVIEW

File: numbat-cli/src/main.rs [hunk 1]
```diff
@@ -315,10 +315,21 @@ impl Cli {
         let result = self.repl_loop(&mut rl, interactive);
 
         if interactive {
-            rl.save_history(&history_path).context(format!(
-                "Error while saving history to '{}'",
-                history_path.to_string_lossy()
-            ))?;
+            // Try to save the history, and if it fails because prerequisite directories don't exist, create them.
+            if let Err(e) = rl.save_history(&history_path) {
```
Comment (sharkdp) [inline numbat-cli/src/main.rs:319]: Instead of checking whether it fails, we could always call `create_dir_all`.

Comment (hamirmahal) [reply numbat-cli/src/main.rs]: Let me know if the latest changes are more or less what you had in mind.

Comment (hamirmahal) [reply numbat-cli/src/main.rs]: Let me know if the latest changes are more or less what you had in mind.

**Comment by sharkdp** [comment]
We actually have a call to `fs::create_dir` in `get_history_path`. The problem seems to be that you didn't have a `~/.local/share` folder at all? Is that possible?

I changed the fix to use `fs::create_dir_all` instead.

**Review by sharkdp — APPROVED** [review]

**Pull request merged**